### PR TITLE
Use JSON API for addNews()

### PR DESCRIPTION
**Pull request by AnnikaCodes**
As I discussed with Zarel on Discord.

### CODE REVIEW

File: js/client-mainmenu.js [hunk 1]
```diff
@@ -124,14 +124,34 @@
...
+					for (var i in [0, 1]) {
+						var post = data[i];
+						var hasRead = data[i].id && Dex.prefs('readnews') === '' + data[i].id;
+
+						html += '<div class="newsentry' + (hasRead ? '' : ' unread') + '">';
+						if (post.title) html += '<h4>' + post.title + '</h4>';
+						if (post.summaryHTML) html += '<p>' + post.summaryHTML + '</p>';
+						html += '<p>';
+						if (post.author) html += '—<strong>' + post.author + '</strong>';
+						if (post.date && !isNaN(parseInt(post.date, 10))) {
```
Comment (Zarel) [inline js/client-mainmenu.js:142]: Is `post.date` not already a number? That's a bug; I'll fix it.

Comment (AnnikaCodes) [reply js/client-mainmenu.js:142]: In the JSON sent by https://pokemonshowdown.com/news.json, it's a string.

Comment (Zarel) [reply js/client-mainmenu.js:142]: It's a number now.

File: js/client-mainmenu.js [hunk 2]
```diff
@@ -124,14 +124,34 @@
...
+							html += '<small class="date"> on ' + new Date(parseInt(post.date, 10) * 1000).toDateString() + '</small>';
```
Comment (Zarel) [inline js/client-mainmenu.js:143]:
```suggestion
						if (post.date) {
							html += '<small class="date"> on ' + new Date(post.date * 1000).toDateString() + '</small>';
```

File: js/client-mainmenu.js [hunk 3]
```diff
@@ -124,14 +124,34 @@
...
+					}
+					this.addPseudoPM({
+						title: 'Latest News',
+						html: html,
+						attributes: 'data-newsid="' + (data[0].id ? data[0].id : '1990') + '"',
+						cssClass: 'news-embed',
+						maxHeight: 'none'
+					});
+				}
 			});
```
Comment (Zarel) [inline js/client-mainmenu.js:155]:
```suggestion
			var self = this;
			$.ajax({
				dataType: "json",
				url: "https://pokemonshowdown.com/news.json",
				success: function (data) {
					var html = '';
					for (var i in [0, 1]) {
						var post = data[i];
						var hasRead = data[i].id && Dex.prefs('readnews') === '' + data[i].id;

						html += '<div class="newsentry' + (hasRead ? '' : ' unread') + '">';
						if (post.title) html += '<h4>' + post.title + '</h4>';
						if (post.summaryHTML) html += '<p>' + post.summaryHTML + '</p>';
						html += '<p>';
						if (post.author) html += '—<strong>' + post.author + '</strong>';
						if (post.date) {
							html += '<small class="date"> on ' + new Date(post.date * 1000).toDateString() + '</small>';
						}
						html += '</p></div>';
					}
					self.addPseudoPM({
						title: 'Latest News',
						html: html,
						attributes: 'data-newsid="' + (data[0].id ? data[0].id : '1990') + '"',
						cssClass: 'news-embed',
						maxHeight: 'none'
					});
				}
			});
```

**Comment by Zarel** [comment]
Looks good, thanks!

File: js/client-mainmenu.js [hunk 4]
```diff
@@ -124,14 +124,34 @@
...
-				attributes: 'data-newsid="' + newsId + '"',
-				cssClass: 'news-embed',
-				height: 400
+			var self = this;
+			$.ajax({
+				url: "https://pokemonshowdown.com/news.json",
+				dataType: "json",
+				success: function (data) {
+					var html = '';
+					for (var i in [0, 1]) {
```
Comment (Zarel) [inline js/client-mainmenu.js:133]: Wait, shit, this should be fixed; it works but all the guides online say never to do it.

Comment (Zarel) [reply js/client-mainmenu.js:133]: Also it breaks older browsers because it's explicitly incompatible with our polyfill.

Comment (AnnikaCodes) [reply js/client-mainmenu.js:133]: Oh, my bad.

Comment (Zarel) [reply js/client-mainmenu.js:133]: Don't worry, I already fixed it. This was something I was going to fix earlier but forgot.

**Pull request merged**